buy_together_app/models.py

```diff
@@ -1,3 +1,29 @@
-# from django.db import models
+from django.db import models
 
-# Create your models here.
+
+class Product(models.Model):
+    qr_code = models.CharField(max_length=32, primary_key=True)
+    product_name = models.CharField(max_length=32)
+    description = models.TextField()
+
+    @staticmethod
+    def filter_qr(qr):
+        if qr is not None:
+            return Product.objects.filter(qr_code=qr)
+        return Product.objects.all()
+
+    @staticmethod
+    def filter_name(nm):
+        if nm is not None and nm != "":
```

Comment:
I think this will be cleaner.
```suggestion
        if nm:
```

Reply:
What if the product name is an empty string. Can't you filter on that as well?

Reply:
Anything preventing from adding to the db a product with an empty name?

Reply:
Yes that will be in the forms in the front end,we can add that here too,what is your idea?

Reply:
Done adding validator to have min length 1

Reply:
i suggest enforcing it at the db level so that users with empty names cannot be added to the db - for example when bypassing the forms.

See the discussion here: https://forum.djangoproject.com/t/django-change-user-models-minimum-length/2595/5

Reply:
Done,for both names and descriptions

Reply:
Cool. Now also add tests to verify the validation :)

Reply:
Done